### PR TITLE
Adding CMK changes for PS cmdlets

### PR DESCRIPTION
**Pull request by viparek**
## Description

This PR contains changes for adding additional parameters in the TransparentDataEncryption, ManagedInstanceTransparentDataEncryption, Server and ManagedInstance cmdlets for CMK improvements.

New params added:
- AutoRotationEnabled
- PrimaryUserAssignedIdentities
- KeyId

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Sql/Sql.LegacySdk/Generated/Models/EncryptionProtectorCreateOrUpdateProperties.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public string ServerKeyType
             get { return this._serverKeyType; }
             set { this._serverKeyType = value; }
         }
-
```
Comment (viparek) [inline src/Sql/Sql.LegacySdk/Generated/Models/EncryptionProtectorCreateOrUpdateProperties.cs:56]: Will remove this blank space

File: src/Sql/Sql/Sql.csproj [hunk 2]
```diff
@@ -24,6 +24,10 @@
     <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.48.0-preview" />
     <PackageReference Include="System.Security.Permissions" Version="4.5.0" />
   </ItemGroup>
+  
+  <ItemGroup>
```
Comment (viparek) [inline src/Sql/Sql/Sql.csproj:28]: This was added for testing with local sdk changes. I will remove this.

File: src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs [hunk 3]
```diff
@@ -182,6 +182,20 @@ public class SetAzureSqlManagedInstance : ManagedInstanceCmdletBase
         [PSArgumentCompleter("None", "1.0", "1.1", "1.2")]
         public string MinimalTlsVersion { get; set; }
 
+        /// <summary>
+        /// Id of the primary user assigned identity
```
Comment (elnatad) [inline src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs:186]: this should be "User Managed Identity (UMI))" Lets do the same in other places.

File: src/Sql/Sql.LegacySdk/Generated/Models/EncryptionProtectorProperties.cs [hunk 4]
```diff
@@ -62,7 +62,19 @@ public string Uri
...
         }
-        
+
+        private bool? _isAutoRotationEnabled;
+
+        /// <summary>
+        /// Optional. Gets or sets the Azure Sql Server Encryption
+        /// Protector Key Rotation Status
+        /// </summary>
+        public bool? AutoRotationEnabled
```
[the remark this answers is not in the document]
Comment (viparek) [reply src/Sql/Sql.LegacySdk/Generated/Models/EncryptionProtectorProperties.cs]: Fixed

File: src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs [hunk 5]
```diff
@@ -323,6 +323,20 @@ public class NewAzureSqlManagedInstance : ManagedInstanceCmdletBase
...
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The primary user assigned identity id")]
+        public string PrimaryUserAssignedIdentityId { get; set; }
+
+        /// <summary>
+        /// URI of the key to use for encryption
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "URI of the key to use for encryption")]
```
[the remark this answers is not in the document]
Comment (viparek) [reply src/Sql/Sql/ManagedInstance/Cmdlet/NewAzureSqlManagedInstance.cs]: Fixed

File: src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs (hunk 3 above)
Comment (viparek) [reply src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs:186]: Fixed here and all other places

File: src/Sql/Sql/Common/ResourceIdentityHelper.cs [hunk 6]
```diff
@@ -12,25 +12,41 @@
...
+
+            identityResult = new Management.Sql.Models.ResourceIdentity()
+            {
+                Type = ResourceIdentityType.SystemAssigned.ToString()
+            };
+
+            return identityResult;
+        }
+
+        public static Management.Sql.Models.ResourceIdentity GetUserAssignedIdentity(List<string> userAssignedIdentities)
```
Comment (alswanso) [inline src/Sql/Sql/Common/ResourceIdentityHelper.cs:40]: You could probably combine these function into one function

File: src/Sql/Sql/Common/ResourceIdentityHelper.cs [hunk 7]
```diff
@@ -12,25 +12,41 @@
...
+            identityResult = new Management.Sql.Models.ResourceIdentity()
             {
-                identityResult = new Management.Sql.Models.ResourceIdentity()
-                {
-                    Type = ResourceIdentityType.SystemAssigned.ToString()
-                };
-            }
+                Type = ResourceIdentityType.UserAssigned.ToString(),
+                // TODO 
+                // Add user assigned identities.
```
Comment (alswanso) [inline src/Sql/Sql/Common/ResourceIdentityHelper.cs:48]: Please set the UserIdentites properties to a dictionary where the keys are the user resource ids the user passed and the object is just a default UserIdentity object

Comment (viparek) [reply src/Sql/Sql/Common/ResourceIdentityHelper.cs]: Done

File: src/Sql/Sql/Common/ResourceIdentityHelper.cs (hunk 6 above)
Comment (viparek) [reply src/Sql/Sql/Common/ResourceIdentityHelper.cs]: Combined the functions into one

**Review by alswanso — DISMISSED** [review]

**Comment by msJinLei** [comment]
Please also fix the test cases failure
```
2021-06-02T00:09:59.8813441Z    ============================ HTTP REQUEST ============================
2021-06-02T00:09:59.8813937Z    
2021-06-02T00:09:59.8814389Z    HTTP Method:
2021-06-02T00:09:59.8814819Z    GET
2021-06-02T00:09:59.8815247Z    
2021-06-02T00:09:59.8815703Z    Absolute Uri:
2021-06-02T00:09:59.8816441Z    https://management.azure.com/subscriptions/2c647056-bab2-4175-b172-493ff049eb29/resourceGroups/sql-dm-cmdlet-test-rgps3877/providers/Microsoft.Sql/servers/sql-dm-cmdlet-serverps3877?api-version=2020-11-01-preview
2021-06-02T00:09:59.8817158Z    
2021-06-02T00:09:59.8817598Z    Headers:
2021-06-02T00:09:59.8818130Z    x-ms-client-request-id        : 37a1eebc-f225-480b-a8fc-41acb55105f2
2021-06-02T00:09:59.8818742Z    Accept-Language               : en-US
2021-06-02T00:09:59.8819170Z    
2021-06-02T00:09:59.8819562Z    Body:
2021-06-02T00:09:59.8819927Z    
2021-06-02T00:09:59.8820283Z    
2021-06-02T00:09:59.8820661Z    
2021-06-02T00:09:59.8821022Z   
2021-06-02T00:09:59.8821558Z    ============================ HTTP RESPONSE ============================
2021-06-02T00:09:59.8822017Z    
2021-06-02T00:09:59.8822386Z    Status Code:
2021-06-02T00:09:59.8822841Z    NotFound
2021-06-02T00:09:59.8823214Z    
2021-06-02T00:09:59.8823587Z    Headers:
2021-06-02T00:09:59.8824043Z    Cache-Control                 : no-cache
2021-06-02T00:09:59.8824531Z    Pragma                        : no-cache
2021-06-02T00:09:59.8825033Z    x-ms-failure-cause            : gateway
2021-06-02T00:09:59.8825573Z    x-ms-request-id               : 79df4d2e-2e0f-4146-9fd3-aa5bbb9e42bc
2021-06-02T00:09:59.8826143Z    x-ms-correlation-request-id   : 79df4d2e-2e0f-4146-9fd3-aa5bbb9e42bc
2021-06-02T00:09:59.8826772Z    x-ms-routing-request-id       : WESTUS:20210516T220611Z:79df4d2e-2e0f-4146-9fd3-aa5bbb9e42bc
2021-06-02T00:09:59.8827381Z    Strict-Transport-Security     : max-age=31536000; includeSubDomains
2021-06-02T00:09:59.8827919Z    X-Content-Type-Options        : nosniff
2021-06-02T00:09:59.8828464Z    Date                          : Sun, 16 May 2021 22:06:10 GMT
2021-06-02T00:09:59.8828985Z    azSdkTestPlayBackMode         : true
2021-06-02T00:09:59.8829412Z    
2021-06-02T00:09:59.8829784Z    Body:
2021-06-02T00:09:59.8830146Z    {
2021-06-02T00:09:59.8830535Z      "error": {
2021-06-02T00:09:59.8830957Z        "code": "ResourceNotFound",
2021-06-02T00:09:59.8831738Z        "message": "The Resource 'Microsoft.Sql/servers/sql-dm-cmdlet-serverps3877' under resource group 'sql-dm-cmdlet-test-rgps3877' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
2021-06-02T00:09:59.8832463Z      }
2021-06-02T00:09:59.8832866Z    }
2021-06-02T00:09:59.8833234Z   
```

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please
- update the help doc for the cmdlets you change
- describe the change in the ChangeLog.md

**Comment by msJinLei** [comment]
@viparek Please regenerate the help doc for the cmdlets you make changes to

File: src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.ps1 [hunk 8]
```diff
@@ -154,3 +154,41 @@ function Test-SetTransparentDataEncryptionProtector
 		Remove-ResourceGroupForTest $rg
 	}
 }
+
+<#
+	.SYNOPSIS
+	Tests Setting a server transparent data encryption protector
+#>
+function Test-SetTransparentDataEncryptionProtectorWithKeyRotation
```
Comment (msJinLei) [inline src/Sql/Sql.Test/ScenarioTests/TransparentDataEncryptionCrudTests.ps1:162]: do you call this function in the TransparentDataEncryptionCrudTests.cs?

**Review by msJinLei — CHANGES_REQUESTED** [review]

**Comment by viparek** [comment]
Updated the changelog with all the changes. Also modified the help files for all the cmdlets changed.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**